### PR TITLE
Add docstrings for configs

### PR DESCRIPTION
**Pull request by rohanshah18**
## Problem

Docstrings for configs are missing.

## Solution

Added docstrings for PineconeConnection and PineconeConfig classes. PineconeConnection is mainly expected to be used by users if they are trying to set the custom gRPC channels.

## Type of Change

- [X] Non-code change (docs, etc)

## Test Plan
NA

### CODE REVIEW

File: src/main/java/io/pinecone/configs/PineconeConfig.java [hunk 1]
```diff
@@ -2,52 +2,110 @@
...
 import io.grpc.ManagedChannel;
 import io.pinecone.exceptions.PineconeConfigurationException;
-import io.pinecone.exceptions.PineconeValidationException;
-
+/**
+ * The {@link PineconeConfig} class is responsible for managing the configuration settings
+ * required to interact with the Pinecone API. It provides methods to set and retrieve
+ * the necessary API key, host, source tag, and custom managed channel.
+ */
 public class PineconeConfig {
```
Comment (ssmith-pc) [inline src/main/java/io/pinecone/configs/PineconeConfig.java:10]: users won't construct this directly, right? If not, then same comment as here applies: 
https://github.com/pinecone-io/pinecone-java-client/pull/109#discussion_r1561652707

File: src/main/java/io/pinecone/configs/PineconeConnection.java [hunk 2]
```diff
@@ -16,8 +16,8 @@
 import java.util.concurrent.TimeUnit;
 
 /**
- * Handles communication with a Pinecone service or router. One PineconeConnection can be shared
- * and used concurrently by multiple threads.
+ * The {@link PineconeConnection} class handles communication with a Pinecone service or router. One PineconeConnection
+ * can be shared and used concurrently by multiple threads.
```
Comment (ssmith-pc) [inline src/main/java/io/pinecone/configs/PineconeConnection.java:20]: And same comment here, I think?

File: src/main/java/io/pinecone/configs/PineconeConfig.java (hunk 1 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/configs/PineconeConfig.java:10]: users may construct this if they are trying to set the gRPC channel and pass it to the PineconeConnection constructor to get gRPC stub

File: src/main/java/io/pinecone/configs/PineconeConnection.java (hunk 2 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/configs/PineconeConnection.java:20]: https://github.com/pinecone-io/pinecone-java-client/pull/110/files#r1561727854

File: src/main/java/io/pinecone/configs/PineconeConfig.java (hunk 1 above)
Comment (rohanshah18) [reply src/main/java/io/pinecone/configs/PineconeConfig.java:10]:
```
import io.grpc.ManagedChannel;
import io.grpc.netty.GrpcSslContexts;
import io.grpc.netty.NegotiationType;
import io.grpc.netty.NettyChannelBuilder;
import io.pinecone.clients.AsyncIndex;
import io.pinecone.clients.Index;
import io.pinecone.configs.PineconeConfig;
import io.pinecone.configs.PineconeConnection;
import io.pinecone.exceptions.PineconeException;

import javax.net.ssl.SSLException;
import java.util.concurrent.TimeUnit;
...


PineconeConfig config = new PineconeConfig("apikey");
String endpoint = "some-endpoint";
NettyChannelBuilder builder = NettyChannelBuilder.forTarget(endpoint);

// Custom channel with timeouts
try {
    builder = builder.overrideAuthority(endpoint)
            .negotiationType(NegotiationType.TLS)
            .keepAliveTimeout(5, TimeUnit.SECONDS)
            .sslContext(GrpcSslContexts.forClient().build());
} catch (SSLException e) {
    throw new PineconeException("SSL error opening gRPC channel", e);
}

// Build the managed channel with the configured options
ManagedChannel channel = builder.build();
config.setCustomManagedChannel(channel);

// Construct the connection object
PineconeConnection connection = new PineconeConnection(config);

// Construct the index object for synchronous data plane operations
Index index = new Index(connection, "example-index");
index.describeIndexStats();

// Construct the async index object for asynchronous data plane operations
AsyncIndex asyncIndex = new AsyncIndex(connection, "example-index");
asyncIndex.describeIndexStats();
```

Is it okay to include this example? Its end to end construction of index or async index using PineconeConfig and PineconeConnection with custom gRPC channel but I do feel like its long

Comment (rohanshah18) [reply src/main/java/io/pinecone/configs/PineconeConfig.java]: Done!

File: src/main/java/io/pinecone/configs/PineconeConfig.java [hunk 3]
```diff
@@ -2,52 +2,143 @@
 
 import io.grpc.ManagedChannel;
 import io.pinecone.exceptions.PineconeConfigurationException;
-import io.pinecone.exceptions.PineconeValidationException;
```
Comment (rohanshah18) [inline src/main/java/io/pinecone/configs/PineconeConfig.java:5]: Was an extra import

**Review by austin-denoble — APPROVED** [review]
Looks good to me, nice work. 🚢

**Pull request merged**